nomad/job_endpoint.go

```diff
@@ -88,6 +88,11 @@ func (j *Job) Register(args *structs.JobRegisterRequest, reply *structs.JobRegis
 		return fmt.Errorf("missing job for registration")
 	}
 
+	// defensive check; http layer and RPC requester should ensure namespaces are set consistently
+	if args.Namespace != args.Job.Namespace {
```

Comment:
There's a RequestNamespace helper for properly defaulting unset args.Namespace to DefaultNamespace.

```suggestion
	if args.RequestNamespace() != args.Job.Namespace {
```

Reply:
Umm.. I'd like to make the stronger claim that args.Namespace is always populated and matches the job's namespace.  At this stage of RPC layer, the job should be rendered completely and no defaulting should take place iiuc.  What do you think?

Reply:
Decided to switch to using `RequestNamespace()` as it's the canonical way to access field and it does feel like consistency is valuable here.  Thanks for catching it.